### PR TITLE
Add support case creation.

### PR DESCRIPTION
**Pull request by fabricematrat**

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/theblues-lib/126/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/theblues-lib/127/
Test PASSed.

**Comment by jrwren** [comment]
👍 

QA worked exactly as stated in email.

File: theblues/support.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    Wishlist = "Wishlist"
+    Low = "Low"
+    Medium = "Medium"
+    High = "High"
+    Critical = "Critical"
+
+
+class Support(object):
+
+    BUSINESS_IMPACT = '00ND0000005lqBV'
```
Comment (bac) [inline theblues/support.py:31]: What is this magic string? Can you add a comment about where it comes from?

File: theblues/support.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    def __init__(self, url, orgId, recordType):
+        """Initializer.
+
+        @param url The url to the Support server.
+        @param orgId the organization Id for SalesForce.
+        @param recordType the record r.
```
Comment (bac) [inline theblues/support.py]: What does "the record r" mean?

File: theblues/support.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        if '' == name or name is None:
+            raise ValueError('empty name')
+        if '' == subject or subject is None:
+            raise ValueError('empty subject')
+        if '' == description or description is None:
+            raise ValueError('empty description')
+        if '' == businessImpact or businessImpact is None:
+            raise ValueError('empty business impact')
+        if priority is None:
+            raise ValueError('empty priority')
```
Comment (bac) [inline theblues/support.py]: Ensure the priority is from the set of known priorities?

File: theblues/tests/test_support.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        with HTTMock(self.create_case):
+            self.support.create_case('someone',
+                                     'someone@email.com',
+                                     'My subject', 'my description',
+                                     'some businessImpact', Priority.L1,
+                                     '4325345345234')
+
+    def test_create_case_empty_name(self):
+        with self.assertRaises(ValueError):
+            self.support.create_case('',
```
Comment (bac) [inline theblues/tests/test_support.py:48]: Can you get the exception message and ensure it is what you expect?

**Comment by bac** [comment]
:+1: with some suggestions.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/theblues-lib/128/
Test PASSed.

**Comment by fabricematrat** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/theblues-lib-merge

**Pull request merged**